txstratum/protocol.py

```diff
@@ -184,6 +186,14 @@ def connection_lost(self, exc: Optional[Exception]) -> None:
         if self.refresh_job_task:
             asyncio.ensure_future(self.refresh_job_task.stop())
 
+    def ask_miner_to_reconnect(self) -> None:
+        """Ask the miner to reconnect. We just want to force it to disconnect with this."""
+        self.log.info('Asking miner to reconnect')
+        self.send_request('client.show_message', ['Will force reconnection'], None)
+        self.send_request('client.reconnect', [], None)
+
+        self.transport.close()
```

Comment:
When `connection_made` is called? `self.transport` is not defined until that. 

Is it possible that this raise an exception depending on the moment it's called?

Reply:
It is called just after an instance of StratumProtocol is created, which happens in the manager:

```
    def __call__(self) -> StratumProtocol:
        """Return an instance of StratumProtocol for a new connection."""
        protocol = StratumProtocol(self)
```

But to be 100% honest with you, this is a mechanism I don't fully understand, because I never found where a Manager instance is called as a callable (which would trigger this __call__)

But by putting some logs I could confirm that `connection_made` is called immediately after the creation of a StratumProtocol instance.

So we will not close it before it is created.